### PR TITLE
Add option for moving statedir into targetdir

### PR DESCRIPTION
**Pull request by gtkramer**
The current install approach favors disk space by allowing for the state
dir to exist outside of the target dir.  However, the two directories
can exist across block device boundaries.  To favor time to install,
adding an option to relocate the state dir into the target dir so they
are located within the same block device.  This can reduce install time
by as much as 60%.  Once the install has completed, the state dir is
removed from the target dir.

This has a couple of side effects.  When this option is used, the
template needs to allocate space for the state dir in addition to actual
content.  Additionally, this option cannot be used in conjunction with
the statedir option, as the state dir in this case is automatically
derived.  Because of these side effects, this option is not enabled by
default.

Signed-off-by: George T Kramer <george.t.kramer@intel.com>

### CODE REVIEW

**Comment by gtkramer** [comment]
I'm not sure if I filled out the list of commands correctly for the added test, copy_os_fast_install_good.  Do I have enough booleans listed?

File: ister.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def run_command(cmd, raise_exception=True, log_output=True, environ=None,
                 LOG.debug(decoded_line)
 
         if proc.poll() and raise_exception:
-            if len(stderr) > 0:
```
Comment (matthewrsj) [inline ister.py]: style fix-up/changes should go in a separate commit from the one that changes or adds functionality.

File: ister.py [hunk 2]
```diff
@@ -412,7 +412,7 @@ def get_current_format():
 def copy_os(args, template, target_dir):
     """Wrapper for running install command
     """
-    LOG.info("Starting swupd. May take several minutes")
```
Comment (matthewrsj) [inline ister.py]: Reason for this? Regardless it should not be in the same commit as a functional change.

File: ister.py [hunk 3]
```diff
@@ -423,6 +423,8 @@ def copy_os(args, template, target_dir):
     swupd_command += " --contenturl={0}".format(args.contenturl)
     swupd_command += " --versionurl={0}".format(args.versionurl)
     swupd_command += " --format={0}".format(args.format)
+    if args.fast_install:
+        args.statedir = "{0}/swupd".format(target_dir)
```
Comment (matthewrsj) [inline ister.py]: This will give you `<target_dir>/swupd` instead of `<target_dir>/var/lib/swupd` which is what I assume you want.

File: ister.py [hunk 4]
```diff
@@ -1435,7 +1443,7 @@ def main():
         install_os(args)
     except Exception as exep:
         LOG.debug("Failed: {}".format(repr(exep)))
-        # todo: Add arg for loglevel to -v
+        # TODO: Add arg for loglevel to -v
```
Comment (matthewrsj) [inline ister.py]: Comment update should be in a different commit as well.

File: ister.py (hunk 3 above)
Comment (matthewrsj) [reply ister.py]: Ah, you remove it later on after install is complete, so it doesn't really matter.

File: ister.py [hunk 5]
```diff
@@ -1418,6 +1423,9 @@ def handle_options():
     parser.add_argument("-S", "--statedir", action="store",
                         default="/var/lib/swupd",
                         help="Path to swupd state dir")
+    parser.add_argument("-F", "--fast-install", action="store_true",
```
Comment (bryteise) [inline ister.py]: To avoid confusing users, please put this in a [mutual-exclusion](https://docs.python.org/3/library/argparse.html#mutual-exclusion) with the statedir argument.

File: ister.py (hunk 4 above)
Comment (bryteise) [reply ister.py]: And do this for all cases of TODO needing to be corrected please!

File: ister.py (hunk 3 above)
Comment (bryteise) [reply ister.py]: But in case we decide there is going to be a swupd directory in / someday I'd probably use tempfile.mkdtemp with a prefix using the date or something to ensure this won't cause problems.

Comment (bryteise) [reply ister.py]: Actually, to simply life in testing, just include the date in the directory name.

File: ister.py (hunk 4 above)
Comment (gtkramer) [reply ister.py]: I grepped through the code base in the repo and this was the only case.

File: ister.py (hunk 5 above)
Comment (gtkramer) [reply ister.py]: I forgot to do this.  I'll update the PR.

**Comment by gtkramer** [comment]
Rebasing before picking up the work again.  Not ready to merge.

**Comment by gtkramer** [comment]
@bryteise @matthewrsj the latest round of changes are ready and have been tested.  This cuts the installation time down for os-clr-on-clr by 55% from 22 minutes to 10 minutes.

**Comment by gtkramer** [comment]
Actually, let's hold off on this.  Let's chat in person before thinking about what's next for this PR.
